docs/content/3.middleware/2.guides/9.kubernetes-probes.md

```diff
@@ -0,0 +1,130 @@
+```ts[server/routes/healthz.ts]
+export default defineEventHandler(() => 'ok');
+```
+
+
+#### Next: App router
+If using Next.js with app router, use [route handlers](https://nextjs.org/docs/app/api-reference/file-conventions/route):
+::steps
+#step-1 
+Create a `[Next app directory]/app/healthz/route.ts` file
+
+#step-2
+Paste the following content inside
+```ts[app/healthz/route.ts]
+import { NextResponse } from 'next/server';
+
+export function GET() {
+  return NextResponse.json({ status: 'ok' }, { status: 200 });
+}
+```
+::
+
+#### Next: Pages router
+If using Next.js with pages router, use [rewrites](https://nextjs.org/docs/pages/api-reference/next-config-js/rewrites) together with [API routes](https://nextjs.org/docs/pages/building-your-application/routing/api-routes):
+
+::steps
+#step-1
+Add the following code to your `next.config.js`:
+```ts{3-10}[next.config.js]
+module.exports = {
+  // ...
+  async rewrites() {
+    return [
+      {
+        source: '/healthz',
+        destination: '/api/healthz',
+      },
+    ];
+  },
+} 
+```
+This is necessary because by default Next's API routes have an `/api` subpath, but we need it to be <nobr>`/healthz`</nobr> and not <nobr>`/api/healthz`</nobr>.
+
+#step-2
+Create a `[Next app directory]/pages/api/healthz.ts` file
+
+#step-3
+Paste the following content inside:
+```ts[pages/api/healthz.ts]
+import { NextApiRequest, NextApiResponse } from 'next';
+
+export default function handler(_req: NextApiRequest, res: NextApiResponse) {
+  res.status(200).send('ok');
+}
+```
+::
+
+## Readiness probes
+
+### The purpose of readiness probes
+
+Applications in Kubernetes are often hosted in such a way that multiple duplicate instances of an application exist side-by-side simultaneously. Such a duplicate instance is called a replica. Readiness probes allow an application replica to temporarily mark itself as unable to serve requests in a Kubernetes cluster. A *liveness* probe can pass while a *readiness* probe fails - meaning that in general, the application is up, but is still waiting for something to happen so that it can serve requests (e.g. waiting for some secondary, dependent service to become online, like Redis cache).
+
+The `/readyz` endpoints of your application is queried automatically by Alokai Cloud every few seconds to check whether requests should be routed to the queried application replica. One such case - where traffic will should stop directed to a replica - is if an application instance is being killed (if it receives a `SIGTERM` signal). If such signal is received, the replica should stop accepting *new* traffic, but wait a few seconds before shutting down to handle any *old* requests that are still being handled by that particular replica.
```

Comment:
endpoints -> endpoint

Reply:
If such a signal is received, the replica should stop accepting new traffic and then wait a few seconds before shutting down to handle any old requests still being handled by that particular replica.

Reply:
I suggest removing "If such signal is received, the replica should stop accepting *new* traffic, but wait a few seconds before shutting down to handle any *old* requests that are still being handled by that particular replica."